--- a/xbmc/interfaces/builtins/GUIBuiltins.cpp
+++ b/xbmc/interfaces/builtins/GUIBuiltins.cpp
@@ -187,11 +187,14 @@
   else
   { // check if shutdown is specified in particular, and get the time for it
     std::string strHeading;
+    std::string strTime;
     if (StringUtils::EqualsNoCase(params[0], "shutdowntimer"))
+    {
       strHeading = g_localizeStrings.Get(20145);
+      strTime = "30"; // default shutdown time = 30 minutes
+    }
     else
       strHeading = g_localizeStrings.Get(13209);
-    std::string strTime;
     if( CGUIDialogNumeric::ShowAndGetNumber(strTime, strHeading) )
       seconds = static_cast<float>(atoi(strTime.c_str())*60);
     else
