--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,7 @@ include(CMakeDependentOption)
 include(cmake/scripts/common/DependencyOptions.cmake)
 
 # general
+option(HEADERS_ONLY       "Skip building" OFF)
 option(VERBOSE            "Enable verbose output?" OFF)
 option(VERBOSE_FIND       "Enable verbose output of library searches only?" OFF)
 option(ENABLE_CLANGTIDY   "Enable clang-tidy support?" OFF)
@@ -170,6 +171,7 @@ core_find_git_rev(APP_SCMID FULL)
 set(AUDIO_BACKENDS_LIST "" CACHE STRING "Available audio backends")
 set(GL_INTERFACES_LIST "" CACHE STRING "Available GL interfaces")
 
+if(NOT HEADERS_ONLY)
 # Dynamically loaded libraries built with the project
 add_custom_target(${APP_NAME_LC}-libraries)
 set(LIBRARY_FILES "" CACHE STRING "" FORCE)
@@ -626,6 +628,7 @@ if(DOXYGEN_FOUND)
                     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/docs/doxygen
                     COMMENT "Generating Doxygen documentation" VERBATIM)
 endif()
+endif()
 
 # link wrapper
 if(FFMPEG_LINK_EXECUTABLE)
--- a/cmake/scripts/linux/ExtraTargets.cmake
+++ b/cmake/scripts/linux/ExtraTargets.cmake
@@ -40,5 +40,7 @@ if("wayland" IN_LIST CORE_PLATFORM_NAME_
   # Dummy target for dependencies
   add_custom_target(generate-wayland-extra-protocols DEPENDS wayland-extra-protocols.hpp)
 
-  add_dependencies(lib${APP_NAME_LC} generate-wayland-extra-protocols)
+  if(NOT HEADERS_ONLY)
+    add_dependencies(lib${APP_NAME_LC} generate-wayland-extra-protocols)
+  endif()
 endif()
--- a/cmake/scripts/linux/Install.cmake
+++ b/cmake/scripts/linux/Install.cmake
@@ -45,6 +45,7 @@ configure_file(${CMAKE_SOURCE_DIR}/tools
 configure_file(${CMAKE_SOURCE_DIR}/tools/Linux/kodi.metainfo.xml.in
                ${CORE_BUILD_DIR}/${APP_PACKAGE}.metainfo.xml @ONLY)
 
+if(NOT HEADERS_ONLY)
 # Install app
 install(TARGETS ${APP_NAME_LC}
         DESTINATION ${libdir}/${APP_NAME_LC}
@@ -164,6 +165,7 @@ if(INTERNAL_TEXTUREPACKER_INSTALLABLE)
           RENAME "${APP_NAME_LC}-TexturePacker"
           COMPONENT kodi-tools-texturepacker)
 endif()
+endif()
 
 # Install kodi-addon-dev headers
 include(${CMAKE_SOURCE_DIR}/xbmc/addons/AddonBindings.cmake)
