diff --git a/Makefile.in b/Makefile.in
index 64e0cb2d8be1..eb2adcd7e7c5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -224,7 +224,7 @@ ZLIBOBJ = \
 
 LIBOBJ    = $(LIBOBJ_W) $(LIBOBJ_NW) $(ZLIB)
 ALLOBJ_W  = $(NASM) $(LIBOBJ_W)
-ALLOBJ    = $(PROGOBJ) $(LIBOBJ)
+ALLOBJ    = $(PROGOBJ) $(LIBOBJ) $(LIBOBJ_DIS)
 SUBDIRS  = stdlib nasmlib include config output asm disasm x86 \
 	   common zlib macros misc
 XSUBDIRS = nsis win test doc editors
@@ -262,7 +262,7 @@ ndisasm$(X): $(NDISASM) $(MANIFEST) $(DISLIB) $(NASMLIB)
 		$(DISLIB) $(NASMLIB) $(LIBS)
 
 # Make sure we have subdirectories set up...
-$(LIBOBJ) $(LIBOBJ_DIS): $(DIRS)
+$(ALLOBJ): $(DIRS)
 
 #-- Begin Generated File Rules --#
 
