--- a/resources/lib/modules/services.py
+++ b/resources/lib/modules/services.py
@@ -477,8 +477,8 @@
         oe.winOeMain.set_wizard_title(oe._(32311))
 
         # Enable samba
-        self.struct['samba']['settings']['samba_autostart']['value'] = '1'
-        self.initialize_samba()
+        #self.struct['samba']['settings']['samba_autostart']['value'] = '1'
+        #self.initialize_samba()
 
         if hasattr(self, 'samba'):
             oe.winOeMain.set_wizard_text(f'{oe._(32313)}[CR][CR]{oe._(32312)}')
