--- a/lib/lwip/lwip/contrib/ports/unix/posixlib/lwipopts.h
+++ b/lib/lwip/lwip/contrib/ports/unix/posixlib/lwipopts.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023 Joan Lledó <jlledom@member.fsf.org>
+ * Copyright (C) 2023 Joan Lled <jlledom@member.fsf.org>
  *
  * Redistribution and use in source and binary forms, with or without modification,
  * are permitted provided that the following conditions are met:
--- a/lib/lwip/lwip/contrib/ports/unix/posixlib/include/posix/inet.h
+++ b/lib/lwip/lwip/contrib/ports/unix/posixlib/include/posix/inet.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023 Joan Lledó <jlledom@member.fsf.org>
+ * Copyright (C) 2023 Joan Lled <jlledom@member.fsf.org>
  *
  * Redistribution and use in source and binary forms, with or without modification,
  * are permitted provided that the following conditions are met:
--- a/lib/lwip/lwip/contrib/ports/unix/posixlib/include/posix/sockets.h
+++ b/lib/lwip/lwip/contrib/ports/unix/posixlib/include/posix/sockets.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023 Joan Lledó <jlledom@member.fsf.org>
+ * Copyright (C) 2023 Joan Lled <jlledom@member.fsf.org>
  *
  * Redistribution and use in source and binary forms, with or without modification,
  * are permitted provided that the following conditions are met:
